neurodamus/io/synapse_reader.py

```diff
@@ -12,78 +12,125 @@
 from neurodamus.utils.pyutils import gen_ranges
 
 
-def _constrained_hill(K_half, y):
-    K_half_fourth = K_half**4
-    y_fourth = y**4
-    return (K_half_fourth + 16) / 16 * y_fourth / (K_half_fourth + y_fourth)
-
-
-class _SynParametersMeta(type):
-    def __init__(cls, name, bases, attrs):
-        type.__init__(cls, name, bases, attrs)
-        # Init public properties of the class
-        assert hasattr(cls, "_synapse_fields"), "Please define _synapse_fields class attr"
-        cls.dtype = np.dtype(
-            {"names": cls._synapse_fields, "formats": ["f8"] * len(cls._synapse_fields)}
-        )
-        cls.empty = np.recarray(0, cls.dtype)
-        if not hasattr(cls, "_optional"):
-            cls._optional = ()
-        # Reserved fields are used to hold extra info, don't come from edge files
-        if not hasattr(cls, "_reserved"):
-            cls._reserved = ()
+class SynapseParameters:
+    """Synapse parameter names and dtypes for numpy recarrays
+    following the SONATA specification.
+
+    For detailed info on the parameters, see:
+    https://sonata-extension.readthedocs.io/en/latest/recipe.html
```

Comment:
Perhaps https://sonata-extension.readthedocs.io/en/latest/sonata_tech.html#edge-file would be an addition or better link?

Reply:
yes, true